### PR TITLE
Loading optimization attributes for get_header_image_tag()

### PR DESCRIPTION
**Pull request by felipeelia**
This PR contains the changes made by @felixarntz in #4772 plus the decoding=async attribute and unit tests.

Trac ticket: https://core.trac.wordpress.org/ticket/58680

---
**This Pull Request is for code review only. Please keep all other discussion in the Trac ticket. Do not merge this Pull Request. See [GitHub Pull Requests for Code Review](https://make.wordpress.org/core/handbook/contribute/git/github-pull-requests-for-code-review/) in the Core Handbook for more details.**

### CODE REVIEW

File: src/wp-includes/media.php [hunk 1]
```diff
@@ -5643,6 +5643,11 @@ function wp_get_loading_optimization_attributes( $tag_name, $attr, $context ) {
 		return $postprocess( $loading_attrs, true );
 	}
 
+	// The custom header image is always expected to be in the header.
+	if ( 'get_header_image_tag' === $context ) {
+		return $postprocess( $loading_attrs, true );
+	}
```
Comment (spacedmonkey) [inline src/wp-includes/media.php:5649]: Wonder if we should just do this?

```
if ( "template_part_{$header_area}" === $context || 'get_header_image_tag' === $context ) {
	return $postprocess( $loading_attrs, true );
}
```

**Review by spacedmonkey — APPROVED** [review]
This PR looks good to me. I think that @felixarntz  @kt-12

File: src/wp-includes/media.php (hunk 1 above)
Comment (swissspidy) [reply src/wp-includes/media.php:5649]: IMHO keeping them separate makes it easier to read & maintain + easier to add the dedicated inline comments explanations.

File: tests/phpunit/tests/theme/customHeader.php [hunk 2]
```diff
@@ -174,6 +174,83 @@ public function test_get_header_image_tag_with_registered_default_image_and_cust
...
+			array(
+				'default-image' => 'http://localhost/default-header.jpg',
+				'width'         => 60,
+				'height'        => 60,
+			)
+		);
+
+		add_filter(
+			'wp_min_priority_img_pixels',
+			function() {
```
Comment (mukeshpanchal27) [inline tests/phpunit/tests/theme/customHeader.php]:
```suggestion
			static function() {
```

Use static closures for minor performance improvement whenever instance access is not needed.

File: tests/phpunit/tests/theme/customHeader.php [hunk 3]
```diff
@@ -174,6 +174,83 @@ public function test_get_header_image_tag_with_registered_default_image_and_cust
...
+			array(
+				'fetchpriority' => '',
+				'decoding'      => '',
+			)
+		);
+		$this->assertStringNotContainsString( ' fetchpriority="high"', $html );
+		$this->assertStringNotContainsString( ' decoding="async"', $html );
+	}
+
+
```
Comment (mukeshpanchal27) [inline tests/phpunit/tests/theme/customHeader.php]:
```suggestion
```

File: src/wp-includes/theme.php [hunk 4]
```diff
@@ -1265,6 +1266,27 @@ function get_header_image_tag( $attr = array() ) {
 		}
 	}
 
+	$attr = array_merge(
+		$attr,
+		wp_get_loading_optimization_attributes( 'img', $attr, 'get_header_image_tag' )
+	);
+
+	// If the default value of `lazy` for the `loading` attribute is overridden
+	// to omit the attribute for this image, ensure it is not included.
```
Comment (mukeshpanchal27) [inline src/wp-includes/theme.php]:
```suggestion
	/*
	 * If the default value of `lazy` for the `loading` attribute is overridden
	 * to omit the attribute for this image, ensure it is not included.
	 */
```

Use multiline comment.

**Review by mukeshpanchal27 — COMMENTED** [review]
Thank you @felipeelia for PR. It's great that the missing attributes for the `get_header_image_tag()` function were added. However, please note that it may not work for certain themes listed below, as they do not utilize the function for the header image and instead rely on custom markup:

- Twenty Eleven ❌
- Twenty Fourteen ❌
- Twenty Sixteen ❌
- Twenty Ten ❌
- Twenty Twelve ❌
- Twenty Seventeen ✅

**Comment by mukeshpanchal27** [comment]
Per https://github.com/WordPress/wordpress-develop/pull/4790#pullrequestreview-1514206718 feedback do we need to open separate ticket for that?

cc. @felixarntz @spacedmonkey

File: src/wp-includes/theme.php (hunk 4 above)
Comment (felipeelia) [reply src/wp-includes/theme.php]: Changed in 24e883e

File: tests/phpunit/tests/theme/customHeader.php (hunk 3 above)
Comment (felipeelia) [reply tests/phpunit/tests/theme/customHeader.php]: Changed in 24e883e

File: tests/phpunit/tests/theme/customHeader.php (hunk 2 above)
Comment (felipeelia) [reply tests/phpunit/tests/theme/customHeader.php]: Awesome, thanks for the tip @mukeshpanchal27. I've changed it in 24e883e.

**Comment by spacedmonkey** [comment]
> Per [#4790 (review)](https://github.com/WordPress/wordpress-develop/pull/4790#pullrequestreview-1514206718) feedback do we need to open separate ticket for that?
> 
> cc. @felixarntz @spacedmonkey

@mukeshpanchal27 Let's handle this in https://core.trac.wordpress.org/ticket/58675 and https://github.com/WordPress/wordpress-develop/pull/4797

**Review by felixarntz — APPROVED** [review]
Thank you for completing the work on this @felipeelia! LGTM ✅

**Review by westonruter — APPROVED** [review]
LGTM

**Comment by felixarntz** [comment]
Committed in https://core.trac.wordpress.org/changeset/56142

**Pull request closed without merging**